Feature products

tgnv opened the pull request:
На сегодняшний момент с точки зрения каталога товаров реализовано:

1. REST сервис для создания, модификации и удаления записей каталога товаров, а также их выборки по id, наименованию товара, диапазону цен и полным списком.
2. Интеграционные тесты указанных операций.

Commit:
add basic goods feature structure

Commit:
add minor naming changes in products feature

tgnv committed:
Addition of Product entity integration tests

tgnv committed:
some test changes

tgnv committed:
Some minor changes

tgnv committed:
Next Minor changes

Victor-IT committed:
Merge branch 'development' into feature-products

SKuznet commented on src/main/java/io/delivery/entity/Product.java:
```diff
@@ -0,0 +1,37 @@
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
+
+    @Column(name = "name")
```
@Column not required here

tgnv committed:
Addition of:
1. ProductCatalogSection entity
2. ProductImage entity
3. ProductCatalogPersistenseModel UML diagram

Modification of Product entity:
1. Added reference to ProductCatalogSection

tgnv committed:
Merge remote-tracking branch 'origin/feature-products' into feature-products

# Conflicts:
#	pom.xml
#	src/main/resources/log4j.properties
